solve_bimodal_test.go

```diff
@@ -748,6 +748,97 @@ var bimodalFixtures = map[string]bimodalFixture{
 			"bar from baz 1.0.0",
 		),
 	},
+	"require package": {
+		ds: []depspec{
+			dsp(mkDepspec("root 0.0.0", "bar 1.0.0"),
+				pkg("root", "foo")),
+			dsp(mkDepspec("foo 1.0.0"),
+				pkg("foo", "bar")),
+			dsp(mkDepspec("bar 1.0.0"),
+				pkg("bar")),
+			dsp(mkDepspec("baz 1.0.0"),
+				pkg("baz")),
+		},
+		require: []string{"baz"},
+		r: mksolution(
+			"foo 1.0.0",
+			"bar 1.0.0",
+			"baz 1.0.0",
+		),
+	},
+	"require subpackage": {
+		ds: []depspec{
+			dsp(mkDepspec("root 0.0.0", "bar 1.0.0"),
+				pkg("root", "foo")),
+			dsp(mkDepspec("foo 1.0.0"),
+				pkg("foo", "bar")),
+			dsp(mkDepspec("bar 1.0.0"),
+				pkg("bar")),
+			dsp(mkDepspec("baz 1.0.0"),
+				pkg("baz", "baz/qux"),
+				pkg("baz/qux")),
+		},
+		require: []string{"baz/qux"},
+		r: mksolution(
+			"foo 1.0.0",
+			"bar 1.0.0",
+			mklp("baz 1.0.0", "baz/qux"),
+		),
+	},
+	"require impossible subpackage": {
```

Comment:
This brings us back to another variant of a familiar question - if an unimported dep is constrained in the manifest, does that constraint still come into effect?

I think it's pretty clear in this case, though: having the requires list is basically equivalent to having an import path, and these are both emerging from the same manifest (so we can assume the same user has control over it), thus the constraint should be applied. And because deps can't express requires, we don't have the same transitivity problem as with normal imports.